src/visual.ts

```diff
@@ -233,46 +256,67 @@ export class TableHeatMap implements IVisual {
         };
     }
 
-    constructor({
-        host,
-        element
-    }: VisualConstructorOptions) {
-        this.host = host;
-
-        this.div = d3Select(element)
-            .append(TableHeatMap.HtmlObjDiv)
-            .classed(TableHeatMap.ClsNameDivTableHeatMap, true);
-
+    constructor(options: VisualConstructorOptions) {
+        this.host = options.host;
+        this.events = options.host.eventService;
+        this.div = d3Select(options.element)
+          .append(TableHeatMap.HtmlObjDiv)
+          .classed(TableHeatMap.ClsNameDivTableHeatMap, true);
+      
         this.svg = this.div
-            .append(TableHeatMap.HtmlObjSvg)
-            .classed(TableHeatMap.ClsNameSvgTableHeatMap, true);
-
+          .append(TableHeatMap.HtmlObjSvg).attr("role", "listbox")
+          .classed(TableHeatMap.ClsNameSvgTableHeatMap, true);
+        
+      
         this.colorHelper = new ColorHelper(this.host.colorPalette);
-        this.localizationManager = host.createLocalizationManager();
-
+        this.localizationManager = this.host.createLocalizationManager();
+      
         this.formattingSettingsService = new FormattingSettingsService(this.localizationManager);
-
+      
         this.tooltipServiceWrapper = createTooltipServiceWrapper(
-            this.host.tooltipService,
-            element);
-
+          this.host.tooltipService,
+          options.element
+        );
+      
         this.selectionManager = this.host.createSelectionManager();
-
+      
         this.handleContextMenu();
-    }
-
-    public update(options: VisualUpdateOptions): void {
+      
+        this.behavior = new HeatMapBehavior();
+        
+      
+        // Initialize interactivity service
+        
+      
+        this.settingsModel = new SettingsModel();
+      
+        this.interactivityService = createInteractivitySelectionService(options.host)
+      }
+      
+      private interactivityService: IInteractivityService<any>;
```

Comment:
please move this private member above the contructor

Reply:
resolved